gp-includes/routes/project.php

```diff
@@ -288,9 +288,13 @@ public function delete_get( $project_path ) {
 
 
 	public function new_get() {
-		$project                    = new GP_Project();
-		$project->active            = 1;
-		$project->parent_project_id = gp_get( 'parent_project_id', null );
+		$project         = new GP_Project();
+		$project->active = 1;
+
+		$parent_project_id = gp_get( 'parent_project_id' );
+
+		// Make sure 'parent_project_id' is a string and convert to int ID. Defaults to null.
+		$project->parent_project_id = is_string( $parent_project_id ) && intval( $parent_project_id ) ? intval( $parent_project_id ) : null;
```

Comment:
Same problem here. I have tested it with these GET parameters:

- `/glotpress/projects/-new/`
- `/glotpress/projects/-new/?parent_project_id=146`
- `/glotpress/projects/-new/?parent_project_id=1ss`
- `/glotpress/projects/-new/?parent_project_id[]=146`
- `/glotpress/projects/-new/?parent_project_id[1]=1ss`
- `/glotpress/projects/-new/?parent_project_id==1234www`
- `/glotpress/projects/-new/?parent_project_id==abxss-_1`

```suggestion
		$project->parent_project_id = filter_var( $parent_project_id, FILTER_VALIDATE_INT ) !== false ? intval( $parent_project_id ) : null;
```

Reply:
I'm making it more simpler with `is_numeric()`.